### ISSUE TITLE
Store & Refresh Social Login Tokens & Pass them to final apps

### ISSUE THREAD

**Issue by tieb62** [goauthentik/authentik#3941]
My instance of authentik accepts login via Microsoft & Discord

I want to use Discord & Microsoft Tokens in my final app

For example to make the user join Discord server (of course the corresponding scope is added to authentik)

If authentik could store tokens (& refresh tokens & expiration date) even if it does not refresh them automatically, if the final app can access those 3 values, it can use the token if it's valid, and refresh it if not (if the refreshing must be done by the final app, an authentik endpoint must be set up for the app to push back to authentik the refreshed token and the new expiration & refresh token

TL;DR;
 - Store Social Login Tokens
 - Refresh them
 - Pass them to final apps

**Comment by petittitouan** [comment]
Bump

**Comment by BeryJu** [comment]
you can reference https://docs.goauthentik.io/integrations/sources/discord/#checking-for-membership-of-a-discord-guild to see how the access/refresh tokens can be accessed through the flow context

**Comment by oytuntez** [comment]
That's very useful! I'll build a machine-only flow that we can even call in a later time to fetch tokens. Thank you!

I also figured connecting non-OpenID oAuth2 providers... it feels like I am getting a hold of using Authentik as an integration authentication platform.

**Comment by dominic-r** [comment]
We're closing this in favor of this parent issue: #25307
